lib/rubocop/cop/style/access_modifier_declarations.rb

```diff
@@ -195,15 +195,27 @@ def allowed?(node)
         def autocorrect(corrector, node)
           case style
           when :group
-            def_nodes = find_corresponding_def_nodes(node)
-            return unless def_nodes.any?
-
-            replace_defs(corrector, node, def_nodes)
+            autocorrect_group_style(corrector, node)
           when :inline
+            autocorrect_inline_style(corrector, node)
+          end
+        end
+
+        def autocorrect_group_style(corrector, node)
+          def_nodes = find_corresponding_def_nodes(node)
+          return unless def_nodes.any?
+
+          replace_defs(corrector, node, def_nodes)
+        end
+
+        def autocorrect_inline_style(corrector, node)
+          if node.parent.begin_type? && node.parent.children.size == 2
```

Comment:
I'd suggest either adding a comment here explaining the need for this check or extracting it into a private helper method.